KnockKnock-iOS/Scenes/BottomSheet/BottomSheetRouter.swift

```diff
@@ -8,45 +8,89 @@
 import UIKit
 
 protocol BottomSheetRouterProtocol: AnyObject {
-  static func createBottomSheet(districtSelectDelegate: DistrictSelectDelegate?, districtsType: DistrictsType?) -> UIViewController
+  var view: BottomSheetViewProtocol? { get set }
 
-  func passCityDataToShopSearch(source: BottomSheetViewProtocol, city: String)
-  func passCountyDataToShopSearch(source: BottomSheetViewProtocol, county: String)
+  static func createBottomSheet(
+    districtSelectDelegate: DistrictSelectDelegate?,
+    districtsType: DistrictsType?,
+    deleteAction: (() -> Void)?,
+    feedData: FeedShare?,
+    isMyPost: Bool?
+  ) -> UIViewController
 
-  func navigateToShopSearch(source: BottomSheetViewProtocol)
+  func navigateToShopSearch()
+  func dismissView(action: (() -> Void)?)
 }
 
 final class BottomSheetRouter: BottomSheetRouterProtocol {
 
-  weak var districtSelectDelegate: DistrictSelectDelegate?
+  weak var view: BottomSheetViewProtocol?
 
   static func createBottomSheet(
     districtSelectDelegate: DistrictSelectDelegate? = nil,
-    districtsType: DistrictsType? = nil
+    districtsType: DistrictsType? = nil,
+    deleteAction: (() -> Void)? = nil,
+    feedData: FeedShare? = nil,
+    isMyPost: Bool? = nil
   ) -> UIViewController {
+
     let view = BottomSheetViewController()
+    let interactor = BottomSheetInteractor()
+    let worker = BottomSheetWorker(kakaoShareManager: KakaoShareManager())
     let router = BottomSheetRouter()
 
-    view.router = router
+    view.interactor = interactor
     view.districtsType = districtsType
-    router.districtSelectDelegate = districtSelectDelegate
+    interactor.router = router
+    interactor.worker = worker
+    router.view = view
+
+    interactor.districtSelectDelegate = districtSelectDelegate
+    interactor.deleteAction = deleteAction
+    interactor.feedData = feedData
+
+    guard let isMyPost = isMyPost else { return view }
+    router.setBottomSheetOptions(isMyPost: isMyPost)
 
     return view
   }
 
-  func passCityDataToShopSearch(source: BottomSheetViewProtocol, city: String) {
-    self.districtSelectDelegate?.fetchSelectedCity(city: city)
-    self.navigateToShopSearch(source: source)
+  private func setBottomSheetOptions(isMyPost: Bool) {
+    guard let view = self.view as? BottomSheetViewController else { return }
+
+    view.do {
+      if isMyPost {
+        $0.setBottomSheetContents(
+          contents: [
+            BottomSheetOption.postDelete.rawValue,
+            BottomSheetOption.postEdit.rawValue
+          ],
+          bottomSheetType: .small
+        )
+
+      } else {
+        $0.setBottomSheetContents(
+          contents: [
+            BottomSheetOption.postReport.rawValue,
+            BottomSheetOption.postShare.rawValue,
+            BottomSheetOption.postHide.rawValue
+          ],
+          bottomSheetType: .medium
+        )
+      }
+      $0.modalPresentationStyle = .overFullScreen
```

Comment:
```suggestion
    var contents: [String] = []
    
    if isMyPost {
      contents = [
        BottomSheetOption.postDelete.rawValue,
        BottomSheetOption.postEdit.rawValue
      ]
    } else {
      contents = [
        BottomSheetOption.postReport.rawValue,
        BottomSheetOption.postShare.rawValue,
        BottomSheetOption.postHide.rawValue
      ]
    }
    
    view.setBottomSheetContents(contents: contents, bottomSheetType: isMyPost ? .small : .medium)
    view.modalPresentationStyle = .overFullScreen
```
그냥 코드 정리이긴 한데 저라면 이렇게 하지 않았을까 싶네요 😁 
최대한 반복적인거 줄이는 ㅎㅎ
이렇게 바꿔달라는 의미는 절대 아닙니다 👍

Reply:
훨씬 보기 좋은 것 같아서 반영했습니다!